### PR TITLE
Fix handling of file attributes in seviri_l1b_nc reader

### LINKED ISSUE #753: seviri l1b netcdf reader needs to be updated due to EUM fixing Attribute  Issue
When creating the seviri l1b netcdf reader i noticed that EUM had incorrectly set some attributes as dimensions. We created a workaround to allow us to work with the netcdf files. 

EUM have now fixed this  but now the reader will fail 

The sevii l1b netcdf reader needs to be updated to now treat these dimension parameters as attributes.

the HRV netcdf metadata is now also correctly formatted and the reader should also be updated to allows us to read this data.

### PR DESCRIPTION
**Pull request by ColinDuff**
EUMETSAT have corrected the netcdf attributes for the combined VIS/IR/HRV Level1b netcdf product.
This PR removes the previous workaround and also reads/displays the HRV data correctly
 Closes #753 

Test for netcdf Level1b data is WIP - as the reader is currently available i thought best to fix it first then add the reader later

No issues raised with flake8

### CODE REVIEW

**Comment by mraspaud** [comment]
@ColinDuff that's great, thanks ! Will this work also with older files from the archive ?

**Comment by ColinDuff** [comment]
Hi Martin - no it will not work with older files that users may have ordered and are stored locally.
 
I assume any netcdf level1b file ordered from EUM now will have the correct attributes . So they may have to reorder any files they have issues with.

**Comment by ColinDuff** [comment]
In response to my assumption re older seviri Level1b netcdf files I tested succesfully on 2 seviri level1b netcdf files ordered from 2017.

File: satpy/readers/seviri_l1b_nc.py [hunk 1]
```diff
@@ -159,49 +166,44 @@ def get_area_def(self, dataset_id):
...
+
+        column_step = self.nc.attrs['vis_ir_column_dir_grid_step'] * 1000.0
+
+        line_step = self.nc.attrs['vis_ir_line_dir_grid_step'] * 1000.0
+        # section 3.1.4.2 of MSG Level 1.5 Image Data Format Description
+        earth_model = int(self.nc.attrs['type_of_earth_model'], 16)
+        if earth_model == 2:
+            ns_offset = 0  # north +ve
+            we_offset = 0  # west +ve
+        elif earth_model == 1:
```
Comment (sfinkens) [inline satpy/readers/seviri_l1b_nc.py]: Maybe add a short comment that the data is shifted to N-W and to correct for it, we have to adjust the corners in the opposite direction, i.e. S-E

File: satpy/etc/readers/seviri_l1b_nc.yaml [hunk 2]
```diff
@@ -5,12 +5,9 @@ reader:
   reader: !!python/name:satpy.readers.yaml_reader.FileYAMLReader
 
 file_types:
-    nc_seviri_l1b:
+    seviri_l1b_nc:
         file_reader: !!python/name:satpy.readers.seviri_l1b_nc.NCSEVIRIFileHandler
-        file_patterns: ['W_XX-EUMETSAT-Darmstadt,VIS+IR+IMAGERY,{satid:4s}+SEVIRI_C_EUMG_{processing_time:%Y%m%d%H%M%S}.nc']
```
Comment (sfinkens) [inline satpy/etc/readers/seviri_l1b_nc.yaml]: Isn't the `VIS+IR+IMAGERY` pattern still needed if one de-selects HRV in the order (EUM data centre)?

**Comment by mraspaud** [comment]
Thanks for checking this out.

Having an open issue for the tests would be nice. Otherwise I can merge this when we get a green light from @sfinkens (and maybe @sjoro if he wants to participate).

File: satpy/readers/seviri_l1b_nc.py (hunk 1 above)
Comment (sjoro) [reply satpy/readers/seviri_l1b_nc.py]: the comment can be copied from the `seviri_l1b_native`-reader.

File: satpy/readers/seviri_l1b_nc.py [hunk 3]
```diff
@@ -159,49 +166,44 @@ def get_area_def(self, dataset_id):
         return area
 
     def get_area_extent(self, dsid):
-        if dsid.name != 'HRV':
-
-            # following calculations assume grid origin is south-east corner
-            # section 7.2.4 of MSG Level 1.5 Image Data Format Description
```
Comment (sjoro) [inline satpy/readers/seviri_l1b_nc.py]: please add the links to the relevant documentation in the docstring (see `seviri_l1b_native`). check that the links still work, too.

File: satpy/etc/readers/seviri_l1b_nc.yaml (hunk 2 above)
Comment (ColinDuff) [reply satpy/etc/readers/seviri_l1b_nc.yaml]: hi, when i ordered the data the Netcdf option didnt allow me to deselect any channels. But now I see that the Netcdf (GSICS) option does so i will update the yaml and test this.

File: satpy/readers/seviri_l1b_nc.py (hunk 1 above)
Comment (ColinDuff) [reply satpy/readers/seviri_l1b_nc.py]: Working on creating tests for this reader now

I will add the comment as suggested by sjoro

File: satpy/readers/seviri_l1b_nc.py (hunk 3 above)
Comment (ColinDuff) [reply satpy/readers/seviri_l1b_nc.py]: done

File: satpy/readers/seviri_l1b_nc.py (hunk 1 above)
Comment (ColinDuff) [reply satpy/readers/seviri_l1b_nc.py]: comment added

File: satpy/etc/readers/seviri_l1b_nc.yaml (hunk 2 above)
Comment (ColinDuff) [reply satpy/etc/readers/seviri_l1b_nc.yaml]: so there is an issue with the netcdf data ordered using netcdf (GSICS) as some attributes are missing, Also spacecraft id is missing from the filename. EUM has been informed

Comment (sfinkens) [reply satpy/etc/readers/seviri_l1b_nc.yaml]: Oh, too bad. Shall we wait for them to fix it or merge now and add support for GSICS nc-files later?

Comment (sjoro) [reply satpy/etc/readers/seviri_l1b_nc.yaml]: after talking to @ColinDuff, I suggest we merge this now and deal with GSICS nc-files later. this way users can use the reader for "regular" nc-files.

**Pull request merged**